tools/query_breakdown/src/main/java/com/google/bigquery/Main.java

```diff
@@ -0,0 +1,95 @@
+package com.google.bigquery;
+
+import org.apache.commons.cli.*;
+
+import java.io.IOException;
+import java.util.Comparator;
+
+/**
+ * this file is the main file for the command line tool. The first argument is a txt file
+ * containing the queries. The second argument, which is optional, is a time limit that we
+ * impose on processing a single query. The third argument, which is optional, is a file
+ * that we want to write the results to.
+ */
+public class Main {
+  public static void main(String[] args) {
+    String inputFile = null;
+    int errorLimit = 0;
+    String outputFile = null;
+    CommandLine cl = createCommand(args);
+
+    // if there is an error in parsing the commandline
+    if (cl == null) {
+      System.exit(1);
+    }
+
+    if (cl.hasOption("r")) {
+      inputFile = cl.getOptionValue("r");
+    }
+    if (cl.hasOption("w")) {
+      outputFile = cl.getOptionValue("w");
+    }
+    if (cl.hasOption("l")) {
+      errorLimit = Integer.parseInt( cl.getOptionValue("l"));
+    }
+
+    // this is where we will put the file I/O logic through the input reader.
+    String originalQuery = null;
+    try {
+      originalQuery = InputReader.readInput(inputFile);
+    } catch (IOException e) {
+      System.out.println("there was an I/O error while reading the input");
+    }
+
+    /* this is where we feed in the original query to the UnparseableDetector, which will find
+       all the unparseable components of the query and output them into the output file if
+       specified. Otherwise, it will be autogenerated.
+     */
+    QueryBreakdown.run(originalQuery, outputFile, errorLimit);
+  }
+
+  /**
+   * This is the method that instantiates a CommandLine object for the Apache CLI Interface. It deals with
+   * command line parsing as well as help generation once parsing is unavailable
+   */
+  public static CommandLine createCommand(String[] args) {
+    CommandLineParser parser = new DefaultParser();
+    Options options = createOptions();
+    HelpFormatter help = new HelpFormatter();
+
+    help.setOptionComparator(new Comparator<Option>() {
+      public int compare(Option option1, Option option2) {
```

Comment:
What's the purpose of having this comparator?

Reply:
The purpose of the comparator is to just order the commands in the order I want it, since the default is alphabetical. 

Before: -r -l -w
After: -r -w -l